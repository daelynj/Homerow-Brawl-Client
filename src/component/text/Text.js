import React from 'react';
import Word from './Word';

function Text(props) {
  const generateWords = () => (
    props.words.map((word, index) => (
      <Word
        key = {index}
        wordIndex = {index}
        word = {word}
        currentWordSubstring = {props.currentWordSubstring}
        input = {props.input}
        currentWordIndex = {props.currentWordIndex}
      />
    ))
  )
  
  return (
    <div className="Words">
      {generateWords()}
    </div>
  )
}

export default Text;
