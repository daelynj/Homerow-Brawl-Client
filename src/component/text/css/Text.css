.text {
  font-size: 20px;
  padding: 1%;
  text-align: left;
  border: 5px solid black;
  margin-left: auto;
  margin-right: auto;
  width: 65%;
  word-break: break-word;
}
