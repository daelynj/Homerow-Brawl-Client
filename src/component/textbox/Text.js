import React from 'react';
import Word from './Word';

function Text(props) {
  function generateWords() {
    let words = [];
    for (var i = 0; i < props.text.length; i++) {
      words[i] = renderWord(props.text[i], i);
    }
    return words;
  }

  function renderWord(word, index) {
    return (
      <Word
        key = {index}
        wordLocation = {index}
        word = {word}
        textToCompare = {props.textToCompare}
        input = {props.input}
        currentWord = {props.currentWord}
      />
    )
  }

  return (
    <div className="Words">
      {generateWords()}
    </div>
  )
}

export default Text;
