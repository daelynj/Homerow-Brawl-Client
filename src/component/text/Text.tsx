import * as React from "react";
import { Word } from "./Word";
import "./css/Text.css";

interface Props {
  words: String[];
  currentWordIndex: number;
  currentWordSubstring: string;
  input: string;
}

export const Text = (props: Props) => {
  const generateWords = () =>
    props.words.map((word, index) => (
      <Word
        key={index}
        onCurrentWord={index === props.currentWordIndex}
        onPastWords={index < props.currentWordIndex}
        word={word + " "}
        currentWordSubstring={props.currentWordSubstring}
        input={props.input}
      />
    ));

  return <div className="text">{generateWords()}</div>;
};
