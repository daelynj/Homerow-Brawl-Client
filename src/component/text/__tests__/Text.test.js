import React from 'react';
import Text from '../Text';
import renderer from 'react-test-renderer';

describe("Text", () => {
  it("renders the expected text", () => {
    const tree = renderer.create(
      <Text
        words = {["This", "is", "text"]}
        textToCompare = {""}
        input = {""}
        currentWord = {0}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
