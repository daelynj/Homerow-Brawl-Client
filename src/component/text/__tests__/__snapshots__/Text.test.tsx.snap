// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders the expected text 1`] = `
<div
  className="text"
>
  <span>
    <span
      className="black"
    >
      T
    </span>
    <span
      className="black"
    >
      h
    </span>
    <span
      className="black"
    >
      i
    </span>
    <span
      className="black"
    >
      s
    </span>
    <span
      className="black"
    >
       
    </span>
  </span>
  <span>
    <span
      className="black"
    >
      i
    </span>
    <span
      className="black"
    >
      s
    </span>
    <span
      className="black"
    >
       
    </span>
  </span>
  <span>
    <span
      className="black"
    >
      t
    </span>
    <span
      className="black"
    >
      e
    </span>
    <span
      className="black"
    >
      x
    </span>
    <span
      className="black"
    >
      t
    </span>
    <span
      className="black"
    >
       
    </span>
  </span>
</div>
`;
